spec/lib/migration/safe_migrate_spec.rb

```diff
@@ -71,11 +71,44 @@ def migrate_up(path)
     expect { User.first.username }.not_to raise_error
   end
 
+  it "allows running a migration that creates an index concurrently if it checks if the index exists first" do
+    Migration::SafeMigrate.enable!
+
+    path =
+      File.expand_path "#{Rails.root}/spec/fixtures/db/migrate/create_index_concurrently_safe_activerecord"
+
+    error = nil
+
+    capture_stdout do
+      begin
+        migrate_up(path)
+      rescue StandardError => e
+        error = e
+      end
+    end
+
+    expect(error.cause.cause.message).to include(
+      "CREATE INDEX CONCURRENTLY cannot run inside a transaction block",
+    )
+  end
+
   it "allows running a migration that creates an index concurrently if it drops the index first" do
+    Migration::SafeMigrate.enable!
+
     path = File.expand_path "#{Rails.root}/spec/fixtures/db/migrate/create_index_concurrently_safe"
-    output = capture_stdout { expect do migrate_up(path) end.to raise_error(StandardError) }
+    error = nil
+
+    capture_stdout do
+      begin
+        migrate_up(path)
+      rescue StandardError => e
+        error = e
+      end
+    end
 
-    expect(output).not_to include(described_class::UNSAFE_DROP_INDEX_CONCURRENTLY_WARNING)
```

Comment:
Previously assertion was not reliable because it does not actually ensure that the `CREATE INDEX CONCURRENTLY` statement was executed. With this PR, we now check for the error caused by running `CREATE INDEX CONCURRENTLY` in the migration. Since the `test` env runs in a transaction no matter what, it will always raise the error.